lib/binaries.sh

```diff
@@ -39,17 +39,24 @@ install_yarn() {
   local version=${2:-1.x}
   local number url code resolve_result
 
-  echo "Resolving yarn version $version..."
-  resolve_result=$(resolve yarn "$version" || echo "failed")
+  if [[ -n "$YARN_BINARY_URL" ]]; then
+    url="$YARN_BINARY_URL"
+    echo "Downloading and installing yarn from $url"
+    code=$(curl "$url" -L --silent --fail --retry 5 --retry-max-time 15 -o /tmp/yarn.tar.gz --write-out "%{http_code}")
+  else
+    echo "Resolving yarn version $version..."
+    resolve_result=$(resolve yarn "$version" || echo "failed")
 
-  if [[ "$resolve_result" == "failed" ]]; then
-    fail_bin_install yarn "$version"
-  fi
+    if [[ "$resolve_result" == "failed" ]]; then
+      fail_bin_install yarn "$version"
+    fi
+
+    read -r number url < <(echo "$resolve_result")
 
-  read -r number url < <(echo "$resolve_result")
+    echo "Downloading and installing yarn ($number)"
+    code=$(curl "$url" -L --silent --fail --retry 5 --retry-max-time 15 -o /tmp/yarn.tar.gz --write-out "%{http_code}")
```

Comment:
Can we take this and the same command on Line 45 and run it after the if/else statement? Since it's using the same url variable both times we only need it once.